File: RptToJv.java
package Jvakt;
/*
 * 2022-06-23 V.54 Michael Ekdal		Added getVersion() to get at consistent version throughout all classes.
 */

import java.io.*;
import java.net.*;
import java.util.Properties;

public class RptToJv {
//	public static void main(String[] args ) throws IOException, UnknownHostException {
	public static void main(String[] args )  {

		String version = "RptToJv ";
		version += getVersion()+".54";
		String host = "127.0.0.1";
		int port = 1956; 
		String id = null;
		String status = "OK";
		String body = " ";
		String agent = " ";
		String type = "R";  // repeating
		String prio = "30";  
		String reply = "";  
		InetAddress inet;

		String jvport   = "1956";
		String jvhost   = "127.0.0.1";

		String config = null;
		File configF;
		boolean swTest = false;

		for (int i=0; i<args.length; i++) {
			if (args[i].equalsIgnoreCase("-config")) config = args[++i];
		}

		if (config == null ) 	configF = new File("Jvakt.properties");
		else 					configF = new File(config,"Jvakt.properties");
		System.out.println("Jvakt: "+version);
		System.out.println("-config file Server: "+configF);

		Properties prop = new Properties();
		InputStream input = null;
		try {
			input = new FileInputStream(configF);
			prop.load(input);
			// get the property value and print it out
			jvhost = prop.getProperty("jvhost");
			jvport = prop.getProperty("jvport");
		} catch (IOException ex) {
			System.out.println("Jvakt.properties not found, continues...");
			//			 ex.printStackTrace();
		}
		port = Integer.parseInt(jvport);
		host = jvhost;

		try {
			inet = InetAddress.getLocalHost();
			System.out.println("-- Inet: "+inet);
			agent = inet.toString();
		}
		catch (Exception e) { System.out.println(e);  }

		for (int i=0; i<args.length; i++) {
			if (args[i].equalsIgnoreCase("-host")) host = args[++i];
			else if (args[i].equalsIgnoreCase("-port")) port = Integer.parseInt(args[++i]);
			else if (args[i].equalsIgnoreCase("-id")) id = args[++i];
			else if (args[i].equalsIgnoreCase("-sts")) status = args[++i];
			else if (args[i].equalsIgnoreCase("-body")) body = args[++i];
			else if (args[i].equalsIgnoreCase("-type")) type = args[++i];
			else if (args[i].equalsIgnoreCase("-ok")) status = "OK";
			else if (args[i].equalsIgnoreCase("-err")) status = "ERR";
			else if (args[i].equalsIgnoreCase("-info")) status = "INFO";
			else if (args[i].equalsIgnoreCase("-prio")) prio = args[++i];
			else if (args[i].equalsIgnoreCase("-test")) swTest = true;
		}

		if (args.length < 1 || (id == null && !swTest)) {
			System.out.println("\n"+version);
			System.out.println("by Michael Ekdal Perstorp Sweden.\n");
			System.out.println("-host \t - default is 127.0.0.1");
			System.out.println("-port \t - default is 1956");
			System.out.println("-id  ");
			System.out.println("-ok   \t -> sts=OK");
			System.out.println("-err  \t -> sts=ERR");
			System.out.println("-info \t -> sts=INFO");
			System.out.println("-sts  \t - default is OK");
			System.out.println("-body \t - Any descriptive text");
			System.out.println("-prio \t - default is 30");
			System.out.println("-type \t - R, S, T, I, D");
			System.out.println("-test \t - test the connection without updating the server side status.");
			System.exit(4);
		}

		if (id == null && !swTest) {
			System.out.println(">>> Failure! The -id switch must contain a value! <<<");
			System.exit(8);
		}
		if (!type.toUpperCase().equals("T") && !type.toUpperCase().equals("R") && !type.toUpperCase().equals("I") && !type.toUpperCase().equals("S") && !type.toUpperCase().equals("D") && !type.toUpperCase().equals("P") && !type.equalsIgnoreCase("Active") && !type.equalsIgnoreCase("Dormant")  ) {
			System.out.println(">>> Failure! The type must be R, I, S, T or D <<<");
			System.exit(8);
		}

		if (status.equals("INFO") && type.toUpperCase().equals("R") ) {
			type = "I";
		}


		//	 System.out.println(args[0]+" - "+args[1]);
		Message jmsg = new Message();
		SendMsg jm = new SendMsg(host, port);
		try {
			reply = jm.open();
			System.out.println("Status: open "+reply);
			if (!reply.equalsIgnoreCase("failed") && !swTest) {
				jmsg.setId(id);
				jmsg.setRptsts(status);
				jmsg.setBody(body);
				jmsg.setType(type);
				jmsg.setAgent(agent);
				jmsg.setPrio( Integer.parseInt(prio) );
				//		jm.sendMsg(jmsg);
				if (jm.sendMsg(jmsg)) System.out.println("-- Rpt Delivered --");
				else            	  System.out.println("-- Rpt Failed --");
				jm.close();
			}
			else {
				if (reply.equalsIgnoreCase("failed")) System.out.println("-- Rpt Failed --");
				else System.out.println("-- Test to Jvakt server succeeded - "+reply);
			}
		}
//		catch (java.net.ConnectException e ) {System.out.println("-- Rpt Failed --" + e); }
		catch (NullPointerException npe2 )   {System.out.println("-- Rpt Failed --" + npe2);}

	}     

	static private String getVersion() {
		String version = "0";
		try { 
			Class<?> c1 = Class.forName("Jvakt.Version",false,ClassLoader.getSystemClassLoader());
			Version ver = new Version();
			version = ver.getVersion();
 		} 
		catch (java.lang.ClassNotFoundException ex) {
			version = "?";
		}
		return version;
	}


}

File: RptToJv.cs
using System;
using System.Net;

namespace RptToJv
{
    public class RptToJv
	{
		public static void Main(string[] args)
		{
			string version = "RptToJv C# (2020-NOV-20)";
			string host = "127.0.0.1";
			int port = 1956;
			string id = null;
			string status = "OK";
			string body = " ";
			string agent = " ";
			string type = "R"; // repeating
			string prio = "30";
			string reply = "";

			try
			{
				//Console.WriteLine("-- Inet: " + Dns.GetHostName().ToString());
				agent = Dns.GetHostName().ToString();
			}
			catch (Exception e)
			{
				Console.WriteLine(e);
			} 

			for (int i = 0; i < args.Length; i++)
			{
				if (args[i].Equals("-host", StringComparison.OrdinalIgnoreCase))
				{
					host = args[++i];
				}
				else if (args[i].Equals("-port", StringComparison.OrdinalIgnoreCase))
				{
					port = int.Parse(args[++i]);
				}
				else if (args[i].Equals("-id", StringComparison.OrdinalIgnoreCase))
				{
					id = args[++i];
				}
				else if (args[i].Equals("-sts", StringComparison.OrdinalIgnoreCase))
				{
					status = args[++i];
				}
				else if (args[i].Equals("-body", StringComparison.OrdinalIgnoreCase))
				{
					body = args[++i];
				}
				else if (args[i].Equals("-type", StringComparison.OrdinalIgnoreCase))
				{
					type = args[++i];
				}
				else if (args[i].Equals("-ok", StringComparison.OrdinalIgnoreCase))
				{
					status = "OK";
				}
				else if (args[i].Equals("-err", StringComparison.OrdinalIgnoreCase))
				{
					status = "ERR";
				}
				else if (args[i].Equals("-info", StringComparison.OrdinalIgnoreCase))
				{
					status = "INFO";
				}
				else if (args[i].Equals("-prio", StringComparison.OrdinalIgnoreCase))
				{
					prio = args[++i];
				}
			}
			if (args.Length < 1 || string.ReferenceEquals(id, null))
			{
				Console.WriteLine("\n\n- "+version);
				Console.WriteLine("- by Michael Ekdal Perstorp Sweden.\n");
				Console.WriteLine("-host \t - default is 127.0.0.1");
				Console.WriteLine("-port \t - default is 1956");
				Console.WriteLine("-id ");
				Console.WriteLine("-ok   \t -> sts=OK");
				Console.WriteLine("-err  \t -> sts=ERR");
				Console.WriteLine("-info \t -> sts=INFO");
				Console.WriteLine("-sts  \t - default is OK");
				Console.WriteLine("-body \t - Any descriptive text");
				Console.WriteLine("-prio \t - default is 30");
				Console.WriteLine("-type \t - R, S, T, I, D");
				Environment.Exit(4);
			}
			if (string.ReferenceEquals(id, null))
			{
				Console.WriteLine(">>> Failure! The -id switch must contain a value! <<<");
				Environment.Exit(8);
			}
			if (!type.ToUpper().Equals("T") && !type.ToUpper().Equals("R") && !type.ToUpper().Equals("I") && !type.ToUpper().Equals("S") && !type.ToUpper().Equals("D") && !type.ToUpper().Equals("P") && !type.Equals("Active", StringComparison.OrdinalIgnoreCase) && !type.Equals("Dormant", StringComparison.OrdinalIgnoreCase))
			{
				Console.WriteLine(">>> Failure! The type must be R, I, S, T or D <<<");
				Environment.Exit(8);
			}
			if (status.Equals("INFO") && type.ToUpper().Equals("R"))
			{
				type = "I";
			}
			Message jmsg = new Message();
			SendMsg jm = new SendMsg(host, port);
			try
			{
				reply = jm.open();
				//Console.WriteLine("Status: open " + reply);
				if (!reply.Equals("failed", StringComparison.OrdinalIgnoreCase))
				{
                    jmsg.setId(id);
					jmsg.setRptsts(status);
					//if (body == null) { body = " "; }
					jmsg.setBody(body);
					jmsg.setType(type);
					jmsg.setAgent(agent);
					jmsg.setPrio(int.Parse(prio));
					if (jm.sendMsg(jmsg))
					{
						Console.WriteLine("-- Rpt Delivered --");
					}
					else
					{
						Console.WriteLine("-- Rpt Failed --");
					}
					jm.close();
				}
				else
				{
					Console.WriteLine("-- Rpt Failed --");
				}
			}
            catch (System.Exception e)
			{
				Console.WriteLine("-- Rpt Failed --" + e);
			}
		}
	}
	
	
	// class message
	public class Message
	{
		private string type = " ";
		private string id = "";
		private string rptsts = " ";
		private string body = " ";
		private string agent = " ";
		private int prio = 30;
		// private int len = 0; 

		public virtual bool setType(string type)
		{
			//type = type.replaceAll("[^a-zA-Z]", "");
			if (type.Length > 10)
			{
				type = type.Substring(0, 10);
			}
			this.type = type;
			return true;
		}
		public virtual bool setId(string id)
		{
			id = id.Trim();
			if (id.Length > 255)
			{
				id = id.Substring(0, 255);
			}
			//id = id.replaceAll("[^a-zA-Z0-9.:*_$#-]", "");
			id = id.Replace('å', 'a');
			id = id.Replace('ä', 'a');
			id = id.Replace('ö', 'o');
			id = id.Replace('Å', 'A');
			id = id.Replace('Ä', 'A');
			id = id.Replace('Ö', 'O');
			id = id.Replace(' ', '_');


			string[] separatingStrings = { "<;>" };
			string[] tab = id.Split(separatingStrings, System.StringSplitOptions.RemoveEmptyEntries);
            this.id = tab[0];
			return true;
		}
		public virtual bool setRptsts(string rptsts)
		{
			rptsts = rptsts.Trim();
			if (rptsts.Length > 255)
			{
				rptsts = rptsts.Substring(0, 255);
			}
			//rptsts = rptsts.replaceAll("[^a-zA-Z]", "");
			string[] separatingStrings = { "<;>" };
			string[] tab = rptsts.Split(separatingStrings, System.StringSplitOptions.RemoveEmptyEntries);
			this.rptsts = tab[0];
			if (this.rptsts.ToUpper().StartsWith("ERR", StringComparison.Ordinal))
			{
				this.rptsts = "ERR";
			}
			else if (this.rptsts.ToUpper().StartsWith("OK", StringComparison.Ordinal))
			{
				this.rptsts = "OK";
			}
			else
			{
				this.rptsts = "INFO ";
			}
			return true;
		}
		public virtual bool setBody(string body)
		{
			//		 regex metacharacters: <([{\^-=$!|]})?*+.>
			body = body.Trim();
			if (body.Length > 255)
			{
				body = body.Substring(0, 255);
			}
			if (body.Length == 0) body = " ";
			//body = body.replaceAll("\\\\", "/");
			//body = body.replaceAll("[^a-zA-Z0-9:;_#/><åäöÅÄÖ\"\\,\\.\\!\\?\\*\\$\\)\\(\\-\\=\\{\\}\\]\\[]", " ");
			//body = body.replaceAll(" {2,}", " "); // replace multiple spaces with one
			//string[] tab = body.Split("<;>", 2);

			string[] separatingStrings = { "<;>" };
			string[] tab = body.Split(separatingStrings, System.StringSplitOptions.RemoveEmptyEntries);
			this.body = tab[0];

			return true;
		}
		public virtual bool setAgent(string agent)
		{
			agent = agent.Trim();
			if (agent.Length > 255)
			{
				agent = agent.Substring(0, 255);
			}
			//agent = agent.replaceAll("[^a-zA-Z0-9.:!?;*_$#)(//\"><-=]", " ");
			//string[] tab = agent.Split("<;>", 2);
			//this.agent = tab[0];
			string[] separatingStrings = { "<;>" };
			string[] tab = agent.Split(separatingStrings, System.StringSplitOptions.RemoveEmptyEntries);
			this.agent = tab[0];
			return true;
		}
		public virtual bool setPrio(int prio)
		{
			this.prio = prio;
			return true;
		}

		public virtual bool MsgOk
		{
			get
			{
				return true;
			}
		}

		public virtual string Id
		{
			get
			{
				return id;
			}
		}
		public virtual string Body
		{
			get
			{
				return body;
			}
		}
		public virtual string Rptsts
		{
			get
			{
				return rptsts;
			}
		}
		public virtual string Type
		{
			get
			{
				return type;
			}
		}
		public virtual string Agent
		{
			get
			{
				return agent;
			}
		}
		public virtual int Prio
		{
			get
			{
				return prio;
			}
		}

	}

}